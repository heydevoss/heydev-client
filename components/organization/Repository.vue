<template>
  <div class="box-backdrop">
    <div class="box">
      <header class="box-header">
        <slot name="header">
          <!--{{repository.name}} -->
          Client
        </slot>
      </header>

      <section class="box-description">
        <slot name="description">
            <!-- {{repository.description}} -->
            UNDER CONSTR Uddd fffdd ddddd dddd d dd  ddC  T ION 
        </slot>
       </section>

       <section class="box-languages">
        <slot name="languages">
            <!--   
            {{repoditory.languange}}
            -->
            Javascript | Vue | HTML
        </slot>
       </section>

       <footer class="box-footer">
          <slot name="footer">
              <img width= "20px" height= "20px" src="..\..\static\fork.png" />  5 <pre>  </pre>
              <img width= "18px" height= "18px" src="..\..\static\star.png" /> 10
        </slot>
      </footer>
      
    </div>
  </div>
</template>

<script>
  export default {
    name: 'Repository',
    props: ['repository']
  };
</script>

<style>
  .box-backdrop {
    display: flex;
    width: 300px;
    height: 215px;
  }
  .box {
    overflow-x: hidden;
    overflow-y: hidden;
    background: #FFFFFF;
    overflow-x: auto;
    display: flex;
    flex-direction: column;
  }
  .box-header,
  .box-footer {
    padding: 15px;
    display: flex;
  }
  .box-header {
    border-bottom: 1px solid #eeeeee;
    color: rgb(0, 0, 0);
  }
  .box-footer {
    border-top: 1px solid #eeeeee;
    justify-content: center;
  }
  .box-description {
    position: relative;
    padding: 20px 10px;
    color: #4AAE9B;
    justify-content: right;
  }
  .box-languages {
    position: relative;
    justify-content: center;
    height: 30px;
    color: rgb(58, 107, 109);
    justify-content: center;
    background-color: rgb(210, 245, 245);
  }
</style>