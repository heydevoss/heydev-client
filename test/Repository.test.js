import { mount } from '@vue/test-utils'
import Repository from '@/components/organization/Repository.vue'

describe('Repository', () => {
  const wrapper = mount(Repository)

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders the correct header', () => {
    expect(wrapper.html()).toContain('<header class="box-header">')
  })

  it('renders img', () => {
    expect(wrapper.html()).toContain('<img')
  })

  test('have the img info (fork)', () => {
    expect(wrapper.html().includes('fork')).toBe(true)
  })

  test('have the img info (stars)', () => {
    expect(wrapper.html().includes('star')).toBe(true)
  })
})
