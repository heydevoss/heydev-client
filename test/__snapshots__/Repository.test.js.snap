// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository renders correctly 1`] = `
<div
  class="box-backdrop"
>
  <div
    class="box"
  >
    <header
      class="box-header"
    >
      
        Client
      
    </header>
     
    <section
      class="box-description"
    >
      
        UNDER CONSTR Uddd fffdd ddddd dddd d dd ddC T ION
      
    </section>
     
    <section
      class="box-languages"
    >
      
        Javascript | Vue | HTML
      
    </section>
     
    <footer
      class="box-footer"
    >
      <img
        height="20px"
        src="..\\\\..\\\\static\\\\fork.png"
        width="20px"
      />
       5
        
      <pre />
       
      <img
        height="18px"
        src="..\\\\..\\\\static\\\\star.png"
        width="18px"
      />
       10
      
    </footer>
  </div>
</div>
`;
